### ISSUE TITLE
cmake builder ignores --local flag

### ISSUE THREAD

**Issue by isage** [keplerproject/luarocks#432]
Local installation path are not passed to cmake when --local flag is used.
Basicly, with #431 this makes cmake builder unusable with --local

**Comment by ignacio** [comment]
What do you expect to change? Variables like

**Comment by isage** [comment]
CMake uses CMAKE_INSTALL_PREFIX, and ignores PREFIX
So cmake should be called with `cmake -DCMAKE_INSTALL_PREFIX=<local_path> ..`

**Comment by ignacio** [comment]
Ah! You can do that already. Take a look at [this rockspec](https://github.com/xpol/lua-rapidjson/blob/master/rapidjson-0.2.2-1.rockspec).

Is that what you want?

**Comment by isage** [comment]
Yep, that's it. Wouldn't it be better to support this natively?

**Comment by ignacio** [comment]
Maybe. IIRC, the make backend works the same way. There might be a reason for it which I'm not aware of. @hishamhm ?

**Comment by isage** [comment]
AFAIK, gnu makefiles just take $PREFIX and/or $DESTDIR into account, like this one https://github.com/mpx/lua-cjson/blob/master/Makefile

**Comment by ignacio** [comment]
Yeah. What I don't know is the reasoning behind "having to pass them explicitly to the makefile".

https://github.com/keplerproject/luarocks/wiki/Creating-a-Makefile-that-plays-nice-with-LuaRocks

There is probably a reason for it, which I'm not aware of.

**Comment by hishamhm** [comment]
For Makefile, the reason is that those variables are not standardized (GNU does propose some policies, but they are not strictly followed).

If these variables are standard in CMake, they could be automatic in LuaRocks 3. CMAKE_INSTALL_PREFIX could be passed automatically. Are LUA_INCLUDE_DIR and LUA_LIBRARIES standard in CMake?

**Comment by ignacio** [comment]
I think they are, in the sense that the script FindLua.cmake (included in CMake's installation) define them and other similar scripts tend to use those variables too.

**Comment by isage** [comment]
Yep, and CMAKE_INSTALL_PREFIX is standard cmake variable.

**Comment by ignacio** [comment]
Ok, I've assigned this to me, but if @isage wants to send a PR, even better :)

**Comment by jmrinaldi** [comment]
Is there a solution to this for existing modules? Trying to install torch-hdf5 locally, and having difficulties.
